diff --git a/table/multiget_context.h b/table/multiget_context.h
index 52dcf1b17..a318f9641 100644
--- a/table/multiget_context.h
+++ b/table/multiget_context.h
@@ -130,6 +130,7 @@ class MultiGetContext {
     }
 
     for (size_t iter = 0; iter != num_keys_; ++iter) {
+	assert(iter < MAX_BATCH_SIZE);
       // autovector may not be contiguous storage, so make a copy
       sorted_keys_[iter] = (*sorted_keys)[begin + iter];
       sorted_keys_[iter]->lkey = new (&lookup_key_ptr_[iter])
